File: app/src/main/java/com/example/pam_app/model/BucketEntry.java
package com.example.pam_app.model;

import java.io.Serializable;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;

public class BucketEntry implements Serializable, Entry {
    public final double amount;
    public final Date date;
    public final String comment;
    public final Integer id;
    public final String bucketTitle;

    public BucketEntry(double amount, Date date, String comment) {
        this.amount = amount;
        this.date = date;
        this.comment = comment;
        this.id = null;
        this.bucketTitle = null;
    }

    public BucketEntry(double amount, Date date, String comment, String bucketTitle) {
        this.amount = amount;
        this.date = date;
        this.comment = comment;
        this.id = null;
        this.bucketTitle = bucketTitle;
    }

    public BucketEntry(double amount, Date date, String comment, int id) {
        this.amount = amount;
        this.date = date;
        this.comment = comment;
        this.id = id;
        this.bucketTitle = null;
    }

    @Override
    public String getAmountString() {
        return "$" + amount;
    }

    @Override
    public String getComment() {
        return comment;
    }

    @Override
    public String getDateString() {
        DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd");
        return dateFormat.format(date);
    }

    public Date getDate() {
        return this.date;
    }

    public Double getAmount() {
        return amount;
    }

    public String getBucketTitle() {
        return bucketTitle;
    }
}

File: app/src/main/java/com/example/pam_app/model/BucketEntry.kt
package com.example.pam_app.model

import android.annotation.SuppressLint
import java.io.Serializable
import java.text.DateFormat
import java.text.SimpleDateFormat
import java.util.*

class BucketEntry : Serializable, Entry {
    val amount: Double
    val date: Date?
    override val comment: String?
    val entryId: Int?
    val bucketId: Int?
    val bucketTitle: String?

    constructor(amount: Double, date: Date?, comment: String?, bucketId: Int?) {
        this.amount = amount
        this.date = date
        this.comment = comment
        entryId = null
        this.bucketId = bucketId
        bucketTitle = null
    }

    constructor(amount: Double, date: Date?, comment: String?, bucketTitle: String?, bucketId: Int?) {
        this.amount = amount
        this.date = date
        this.comment = comment
        entryId = null
        this.bucketId = bucketId
        this.bucketTitle = bucketTitle
    }

    constructor(amount: Double, date: Date?, comment: String?, id: Int, bucketId: Int?) {
        this.amount = amount
        this.date = date
        this.comment = comment
        this.entryId = id
        this.bucketId = bucketId
        bucketTitle = null
    }

    override val amountString: String
        get() = "$$amount"

    override val dateString: String
        @SuppressLint("SimpleDateFormat")
        get() {
            val dateFormat: DateFormat = SimpleDateFormat("yyyy/MM/dd")
            @Suppress("NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS")
            return dateFormat.format(date)
        }

}